x/deployment/client/cli/query.go

```diff
@@ -21,67 +20,105 @@ func GetQueryCmd(key string, cdc *codec.Codec) *cobra.Command {
 		RunE:                       client.ValidateCmd,
 	}
 
-	cmd.AddCommand(flags.GetCommands(
-		cmdDeployments(key, cdc),
-		cmdDeployment(key, cdc),
-		getGroupCmd(key, cdc),
-	)...)
+	cmd.AddCommand(
+		cmdDeployments(),
+		cmdDeployment(),
+		getGroupCmd(),
+	)
 
 	return cmd
 }
 
-func cmdDeployments(key string, cdc *codec.Codec) *cobra.Command {
+func cmdDeployments() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "list",
 		Short: "Query for all deployments",
 		Args:  cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx := context.NewCLIContext().WithCodec(cdc)
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadQueryCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
 
-			dfilters, err := DepFiltersFromFlags(cmd.Flags())
+			queryClient := types.NewQueryClient(clientCtx)
+
+			dfilters, state, err := DepFiltersFromFlags(cmd.Flags())
 			if err != nil {
 				return err
 			}
 
-			obj, err := query.NewClient(ctx, key).Deployments(dfilters)
+			// checking state flag
+			stateVal, ok := types.Deployment_State_value[state]
+
+			if (!ok && (state != "")) || state == "invalid" {
+				return ErrStateValue
+			}
+
+			dfilters.State = types.Deployment_State(stateVal)
 
+			pageReq, err := client.ReadPageRequest(cmd.Flags())
 			if err != nil {
 				return err
 			}
-			return ctx.PrintOutput(obj)
+
+			params := &types.QueryDeploymentsRequest{
+				Filters:    dfilters,
+				Pagination: pageReq,
+			}
+
+			res, err := queryClient.Deployments(context.Background(), params)
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintOutput(res)
 		},
 	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	flags.AddPaginationFlagsToCmd(cmd, "deployments")
 	AddDeploymentFilterFlags(cmd.Flags())
+
 	return cmd
 }
 
-func cmdDeployment(key string, cdc *codec.Codec) *cobra.Command {
+func cmdDeployment() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "get",
 		Short: "Query deployment",
 		Args:  cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx := context.NewCLIContext().WithCodec(cdc)
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadQueryCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			queryClient := types.NewQueryClient(clientCtx)
 
-			id, err := DeploymentIDFromFlags(cmd.Flags(), ctx.FromAddress.String())
+			id, err := DeploymentIDFromFlags(cmd.Flags(), "")
```

Comment:
is the `--from` address picked up automatically another way?

Reply:
`--from` flag is not included in queries

Reply:
Yes, from gets picked up in the `init()` function and saved on `context`.